src/config.c

```diff
@@ -170,6 +183,35 @@ nm_config_t *nm_config_parse(char **err_out) {
                 if (!c_arg) RETERR("file %s: line %d: field 3: expected argument, got end of line\n", fn, line_n);
                 else action->arg = strdup(c_arg);
                 nm_config_push_action(&cur_act, action);
+            } else if (!strcmp(c_typ, "generator")) {
+                // type: generator
+                nm_generator_fn_t generate;
+                nm_menu_location_t loc;
+
+                // type: generator - field 2: location
+                char *c_loc = strtrim(strsep(&cur, ":"));
+                if (!c_loc) RETERR("file %s: line %d: field 2: expected location, got end of line", fn, line_n);
+                else if (!strcmp(c_loc, "main"))   loc = NM_MENU_LOCATION_MAIN_MENU;
+                else if (!strcmp(c_loc, "reader")) loc = NM_MENU_LOCATION_READER_MENU;
+                else RETERR("file %s: line %d: field 2: unknown location '%s'", fn, line_n, c_loc);
+
+                // type: generator - field 3: generator
+                char *c_gen = strtrim(strsep(&cur, ":"));
+                if (!c_gen) RETERR("file %s: line %d: field 3: expected generator, got end of line", fn, line_n);
+                #define X(name) else if (!strcmp(c_gen, #name)) generate = NM_GENERATOR(name);
+                NM_GENERATORS
+                #undef X
+                else RETERR("file %s: line %d: field 3: unknown generator '%s'", fn, line_n, c_gen);
+
+                // type: generator - field 4: argument (optional)
+                char *c_arg = strtrim(cur);
+
+                nm_generator_t *gen = calloc(1, sizeof(nm_generator_t));
+                gen->desc = strdup(c_gen);
+                gen->loc = loc;
+                gen->arg = strdup(c_arg ? c_arg : "");
```

Comment:
(Haven't read through the full thing yet, so I may answer my own question later).

Any specific reason for choosing an empty string instead of a NULL pointer? (e.g., are there mandatory <string.h> calls down the line that would crash on a NULL ptr?)

Reply:
`free(NULL);` is safe, which leaves the actual generator function.

Here, for _test, that' `strtol`, which *probably* crashes on NULL pointers.